pkg/scheduler/plugins/cpunormalization/plugin.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpunormalization
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	resourceapi "k8s.io/kubernetes/pkg/api/v1/resource"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config/validation"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	schedutil "k8s.io/kubernetes/pkg/scheduler/util"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/plugins/nodenumaresource"
+)
+
+const (
+	Name = "CPUNormalization"
+
+	preFilterStateKey = "PreFilter" + Name
+)
+
+var (
+	_ framework.EnqueueExtensions = &Plugin{}
+
+	_ framework.PreFilterPlugin = &Plugin{}
+	_ framework.FilterPlugin    = &Plugin{}
+	_ framework.ScorePlugin     = &Plugin{}
+)
+
+var nodeResourceStrategyTypeMap = map[config.ScoringStrategyType]scorer{
+	config.LeastAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                string(config.LeastAllocated),
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+	config.MostAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return &resourceAllocationScorer{
+			Name:                string(config.MostAllocated),
+			scorer:              mostResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
+		}
+	},
+}
+
+// Plugin checks if a node has sufficient CPU resources under CPU normalization.
+// It works like the NodeResourcesFit plugin but with CPU requests of CPUSet Pods amplified on Nodes with CPU normalization.
+type Plugin struct {
+	handle framework.Handle
+	resourceAllocationScorer
+}
+
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
```

Comment:
Do we need to replace the native ResourceFit plugin (and keep compatible with it in the next versions), or use the Transformer mechanism to mutate NodeInfo and SharedPodList?
@eahydra WDYT?

Reply:
I've discussed with @eahydra before and he suggested not to replace the native plugin nor modify the global cache, so I choosed this implementation finally.

Reply:
OK.